### PR TITLE
Populate code for inspec json inheritance

### PR DESCRIPTION
**Pull request by jquick**
Fixes https://github.com/chef/a2/issues/3622

This fix allows the `inspec json` command to pull code info from child profiles when needed.

### CODE REVIEW

**Review by miah — APPROVED** [review]
Thanks @jquick

File: lib/inspec/profile.rb [hunk 1]
```diff
@@ -263,6 +272,7 @@ def info(res = params.dup) # rubocop:disable Metrics/CyclomaticComplexity
 
       res
     end
+    # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```
Comment (jerryaldrichiii) [inline lib/inspec/profile.rb]: I don't think we need this line, when putting it after a method call it is restricted to only that method.

File: lib/inspec/profile.rb [hunk 2]
```diff
@@ -239,6 +239,15 @@ def info(res = params.dup) # rubocop:disable Metrics/CyclomaticComplexity
         data[:impact] = 1.0 if data[:impact] > 1.0
         data[:impact] = 0.0 if data[:impact] < 0.0
         data[:id] = id
+
+        # if the code field is empty try and pull info from dependencies
+        if data[:code].empty? && parent_profile.nil?
+          locked_dependencies.dep_list.each do |_name, dep|
```
Comment (jerryaldrichiii) [inline lib/inspec/profile.rb:245]: Could this be extracted to a method to avoid `Metrics/PerceivedComplexity`?

File: lib/inspec/profile.rb [hunk 3]
```diff
@@ -239,6 +239,15 @@ def info(res = params.dup) # rubocop:disable Metrics/CyclomaticComplexity
...
+            profile = dep.profile
+            code = Inspec::MethodSource.code_at(data[:source_location], profile.source_reader)
+            data[:code] = code unless code.nil? || code.empty?
```
Comment (jerryaldrichiii) [inline lib/inspec/profile.rb:248]: I may just not be following, but won't this get overwritten by the last entry in `dep_list`? Is that the intended behavior?

**Review by jerryaldrichiii — DISMISSED** [review]
Looks great, only a few questions, and a request to remove one line.

File: lib/inspec/profile.rb (hunk 1 above)
Comment (jquick) [reply lib/inspec/profile.rb]: Turns out you can't do it on a method, and its file based.

File: lib/inspec/profile.rb (hunk 3 above)
Comment (jquick) [reply lib/inspec/profile.rb:248]: Yeah we need a break

File: lib/inspec/profile.rb (hunk 2 above)
Comment (jquick) [reply lib/inspec/profile.rb:245]: Tried that, even putting a method call to a new one will blow it. We will need to redo this method in full to break it up more. Due to the nature of this bandaid hot fix we will put it under techDebt.

Comment (jerryaldrichiii) [reply lib/inspec/profile.rb:245]: Understood. No judgment here!

**Review by jerryaldrichiii — APPROVED** [review]

**Pull request merged**